src/otx/core/model/module/classification.py

```diff
@@ -220,14 +229,35 @@ def __init__(
         torch_compile: bool,
         optimizer: list[OptimizerCallable] | OptimizerCallable = lambda p: torch.optim.SGD(p, lr=0.01),
         scheduler: list[LRSchedulerCallable] | LRSchedulerCallable = torch.optim.lr_scheduler.ConstantLR,
+        metric: Metric = HLabelAccuracy,
     ):
         super().__init__(
             otx_model=otx_model,
             torch_compile=torch_compile,
             optimizer=optimizer,
             scheduler=scheduler,
+            metric=metric,
         )
 
+        self.metric = (
+            metric(
+                num_multiclass_heads=self.model.num_multiclass_heads,
+                num_multilabel_classes=self.model.num_multilabel_classes,
+            )
+            if callable(metric)
+            else metric
+        )
+
+        # Temporary, TODO (sungmanc)
+        # OTX can't support the auto-configuration for the H-label classification
```

Comment:
In conclusion, we should support auto-configuration for H-Labels as well. To solve this, we need to add logic to automatically set `num_multiclass_heads`, `num_multilabel_classes`. After that logic is added, we can support auto-configuration.

Reply:
I agree, adding the logic of what you said should be done. I think that could be handled at the next PR